packages/dashboard/src/components/SubwayNav/SubwayNodeBase.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { getIcon, IIconRecord, classNamesFunction, FocusZone, IProcessedStyleSet } from 'office-ui-fabric-react';
+import { ISubwayNavNodeProps, SubwayNavNodeState, ISubwayNavNodeStyleProps, ISubwayNavNodeStyles } from './SubwayNode.types';
+import { getIconMap } from './SubwayNode.styles';
+import { SubwayNode } from './SubwayNode';
+
+const getClassNames = classNamesFunction<ISubwayNavNodeStyleProps, ISubwayNavNodeStyles>();
+
+export class SubwayNodeBase extends React.PureComponent<ISubwayNavNodeProps> {
+  public static defaultProps = {
+    state: SubwayNavNodeState.Current,
+    isSubStep: false,
+    disabled: false
+  };
+
+  private _classNames: IProcessedStyleSet<ISubwayNavNodeStyles>;
+
+  public constructor(props: ISubwayNavNodeProps) {
+    super(props);
+    this._onClickStep = this._onClickStep.bind(this);
+  }
+
+  public render(): JSX.Element {
+    const {
+      styles,
+      isSubStep,
+      disabled,
+      state,
+      subSteps,
+      index,
+      onRenderSubSteps = this._onRenderSubSteps,
+      onRenderStepIcon = this._onRenderStepIcon
+    } = this.props;
+
+    const iconRecord: IIconRecord | undefined = getIcon(getIconMap(this.props.isSubStep!)[this.props.state]);
+
+    this._classNames = getClassNames(styles!, {
+      isSubStep: isSubStep!,
+      disabled: disabled!,
+      state,
+      iconRecord: iconRecord!,
+      hasSubSteps: subSteps ? subSteps.length > 0 : false,
+      index: index!
+    });
+
+    return (
+      <li className={this._classNames.root} data-is-focusable={!this.props.disabled}>
+        <div className={this._classNames.flexContainer} {...!this.props.disabled && { onClick: this._onClickStep }}>
+          {onRenderStepIcon(this.props, this._classNames, iconRecord)}
+          <div className={this._classNames.spacer} />
+          <div className={this._classNames.label}>{this.props.label}</div>
+          <div className={this._classNames.labelSelected}>{this.props.label}</div>

```

Comment:
Seems like all we're doing here is changing the font-weight. Why two elements with hiding/showing?

Reply:
This is because Karina has fade animations between them. So to cross fade I need two elements. That cross fade isn't implemented at the moment so your confusion is valid.

Reply:
Makes sense.